lib/FileUploadDataSource.ts

```diff
@@ -1,13 +1,15 @@
 import {
-  GraphQLDataSourceProcessOptions,
+  type GraphQLDataSourceProcessOptions,
   RemoteGraphQLDataSource,
 } from '@apollo/gateway';
-import { GraphQLResponse } from 'apollo-server-types';
-import { FileUpload, Upload } from 'graphql-upload';
-import { Request, Headers, Response } from 'apollo-server-env';
+// eslint-disable-next-line import/extensions
+import Upload from 'graphql-upload/Upload.js';
+// eslint-disable-next-line import/extensions
+import type { FileUpload } from 'graphql-upload/GraphQLUpload.js';
 import { isObject } from '@apollo/gateway/dist/utilities/predicates';
 import cloneDeep from 'lodash.clonedeep';
 import set from 'lodash.set';
+import type { FetchInterface, FetchOptions } from 'make-fetch-happen';
```

Comment:
Considering what has been suggested [here](https://github.com/profusion/apollo-federation-file-upload/pull/55#issuecomment-1262791533), wouldn't it be better to add `make-fetch-happen` a direct dependency and force it to be used instead of using the default from apollo? (it uses `make-fetch-happen` at the moment, but this can change in future versions).